andreathniah/honeywatt
# PH-8 Containerized C2 servers and database via Docker

### What does this PR do?
This PR containerized C2 servers and databases via docker-compose. It will be able to:
- [x] Add entry to MongoDB via POST request
- [x] Retain MongoDB data even with docker tear down
- [x] Allow MonogoDB's GUI to tunnel into docker

Resolves #8: Containerise C2 servers and database via Docker

### Testing Procedures for Reviewer
#### Test Infrastructure-as-a-Code 
1. Install [Docker](https://docs.docker.com/docker-for-mac/install/), git clone this repo, traverse to `/server`
2. Install npm packages and run the docker environment with the following commands
```
npm install
docker-compose build
docker-compose up -d
```
3. Access `127.0.0.1:80` with your web browser, enter any input
4. Shutdown docker environment
```
docker-compose down
```
5. Restart step (2) again after 15 seconds and you will notice that the data you inputed at step (3) is still there

#### Test Mongo-Docker Tunnel
1. Download [Robo T3](https://robomongo.org/download), ensure that your docker still up and running
```
docker ps
```
2. Create new connection: `localhost:27017`
3. At `docker-node-mongo > Collection > Items`, right-click `View Documents` to view items

### Checklist for Author of PR
- [x] I have completed all functionalities described in the issue 
- [x] I have tested the changes in this pull-request myself 
- [x] I have added tests for my functionality (check if not applicable)
- [x] I have made corresponding changes to the documentation (check if not applicable)
- [x] My codes have been linted with `prettier` (check if not applicable)
- [x] My commit messages follows the guidelines

### Checklist for Reviewers
- [x] The implementation corresponds to the use case which described in the issue
- [x] All new features have appropriate documentation (check if not applicable)
- [x] There are relevant tests added (check if not applicable)
